core/templates/dev/head/filters.js

```diff
@@ -233,6 +233,12 @@ oppia.filter('parameterizeRuleDescription', [
         } else {
           replacementText = inputs[varName];
         }
+        if(typeof replacementText === 'string') {
```

Comment:
Add space before '('.

I think there are some issues with assumptions here. What if there's more than one noninteractive component? Probably should also add some Karma tests for this.

Reply:
What's the problem if more than one noninteractive components exist? For every component, there would be a different replacementText which will be edited by this code.

Reply:
Try adding a multiple choice option with, say, two images. Are both converted correctly or is only the first one handled?

Also, the way this is being done seems a bit ad hoc. Can we restrict the editing to specific interaction types whose rules use HTML? E.g. do something based on the varType?

Reply:
(Also, I still strongly recommend adding some Karma tests.)

Reply:
Just tried the multiple choice interaction, it's working fine.

![pop](https://user-images.githubusercontent.com/22103406/34764495-b91350ee-f614-11e7-852a-4961ee036b59.png)
![pop2](https://user-images.githubusercontent.com/22103406/34764557-ebbd0a9e-f614-11e7-915b-a84b5d98d464.png)

Reply:
Just to be clear, I meant two images in a single option.

Reply:
Isn't the second image, what you are saying? It contains 2 images in a single response.

Reply:
No, I meant 2 images in a single multiple-choice _option_. The contents of a multiple choice option are arbitrary HTML; it's possible for such an HTML box to include multiple components..

Reply:
Just checked it. It's not working in that case. Let me see what can be done here.